mut/redirects/redirect_main.py

```diff
@@ -0,0 +1,396 @@
+"""
+Usage:
+    mut-redirects <source_path> -o <output>
+
+    -h, --help             List CLI prototype, arguments, and options.
+    <source_path>          Path to the file(s) containing redirect rules.
+    -o, --output <output>  File path for the output .htaccess file.
+"""
+
+# Spec URL: https://docs.google.com/document/d/1oI2boFmtzvbbvt-uQawY9k_gLSLbW7LQO2RjVkvtRgg/edit?ts=57caf48b#
+
+import re
+from docopt import docopt
+
+rules = []
+symlinks = []
+definitions = []
+
+class Rule:
+    is_temp = None
+    version = None
+    old_url = None
+    new_url = None
+
+def parse_versions(defs):
+    for x in range (0, len(defs)):
+        if defs[x][0] == 'versions':
+            return defs[x][1]
+
+def generate_rule(is_temp, version, old_url, new_url, is_symlink = False):
+
+    global definitions
```

Comment:
Nix these globals. I'd recommend making a class called something like `RedirectContext` that contains these instance attributes, and anything operating on the context would become a method of that class.